openedx/core/djangoapps/safe_sessions/tests/test_utils.py

```diff
@@ -33,14 +33,16 @@ def assert_regex_not_logged(self, log_string, log_level):
                 yield
 
     @contextmanager
-    def assert_logged_with_message(self, log_string, log_level='error'):
+    def assert_logged_with_message(self, log_substring, log_level='error'):
         """
         Asserts that the logger with the given log_level was called
-        with a string.
+        with a substring.
         """
         with patch('openedx.core.djangoapps.safe_sessions.middleware.log.' + log_level) as mock_log:
             yield
-            mock_log.assert_any_call(log_string)
+            log_messages = [call.args[0] for call in mock_log.call_args_list]
+            assert any(log_substring in msg for msg in log_messages), \
+                f"Expected to find log substring in one of: {log_messages}"
```

Comment:
Was this supposed to be `Expected to find {log_substring}...` ?

Reply:
No, I decided that the readability suffered when I included the value -- and it gets printed out anyway since it's one of the parameters. Here's what it ends up looking like:

```
openedx/core/djangoapps/safe_sessions/tests/test_middleware.py:377: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.8/contextlib.py:120: in __exit__
    next(self.gen)
openedx/core/djangoapps/safe_sessions/tests/test_utils.py:184: in assert_logged_for_both_mismatch
    yield
/usr/lib/python3.8/contextlib.py:120: in __exit__
    next(self.gen)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <openedx.core.djangoapps.safe_sessions.tests.test_middleware.TestSafeSessionMiddleware testMethod=test_warn_on_user_change_in_both>
log_substring = "SafeCookieData user at initial request '1' matches neither user in session: '2' nor user at response time: '3' for request path '/'. Session did not change."
log_level = 'warning'

    @contextmanager
    def assert_logged_with_message(self, log_substring, log_level='error'):
        """
        Asserts that the logger with the given log_level was called
        with a substring.
        """
        with patch('openedx.core.djangoapps.safe_sessions.middleware.log.' + log_level) as mock_log:
            yield
            log_messages = [call.args[0] for call in mock_log.call_args_list]
>           assert any(log_substring in msg for msg in log_messages), \
                f"Expected to find log substring in one of: {log_messages}"
E           AssertionError: Expected to find log substring in one of: ["SafeCookieData user at initial request '1' matches neither user in session: '2' nor user at response time: '3' for request path '/'.\nSession did not change.\nHash of session ID from various sources: cookie_field=43c0cb10; at_request=43c0cb10; at_response=43c0cb10"]
E           assert False
E            +  where False = any(<generator object TestSafeSessionsLogMixin.assert_logged_with_message.<locals>.<genexpr> at 0x7fdf858c6c10>)

openedx/core/djangoapps/safe_sessions/tests/test_utils.py:44: AssertionError
```